app/frontend/src/features/messages/surfing/hostRequestsSlice.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { HostRequest } from "../../../pb/requests_pb";
+
+const initialState = {
+  hostRequests: [] as HostRequest.AsObject[],
+  loading: false,
+  error: null as string | null | undefined,
+};
+
+export type RequestsState = typeof initialState;
+
+export const hostRequestsSlice = createSlice({
```

Comment:
Because I like the actions go up state go down principle. https://redux.js.org/tutorials/fundamentals/part-2-concepts-data-flow . It is also handy when persisting.

Reply:
I also like to keep business logic and state in the model layer.